File: app/src/main/java/com/goldmines/xmlparsing/MainActivity.java
package com.goldmines.xmlparsing;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.TextView;

import com.goldmines.JsonPojos.SimpleQuestion;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import org.json.JSONException;
import org.json.JSONObject;
import org.json.XML;

import java.lang.reflect.Type;

import static com.goldmines.xmlparsing.XMLFile.sampleXml;

public class MainActivity extends AppCompatActivity {
    private TextView tv_json;
    private String jsonString=null ;
    SimpleQuestion questions;



    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tv_json= findViewById(R.id.tv_json);




        JSONObject jsonObj = null;

        try {
            jsonObj = XML.toJSONObject(sampleXml);
        } catch (JSONException e) {
            Log.e("JSON exception", e.getMessage());
            e.printStackTrace();
        }

        Log.d("XML", sampleXml);

        Log.d("JSON", jsonObj.toString());
        /*      tv_json.setText(jsonObj.toString());*/

        jsonString=jsonObj.toString();



        Gson gson = new Gson();

        Type category  = new TypeToken<SimpleQuestion>(){}.getType();

          questions = gson.fromJson(jsonString, category);
        tv_json.setText(questions.getQid());
     /*    *//*   MediaItemArray[] mediaItem = questions.getSimpleQuestion().getMedia().getMediaItemArray();*//*
        String s= "";

       for( MediaItemArray a : mediaItem)
       {
           s += a.toString()+"\n";
       }
        s +="\n\n";*//*
         *//*  AnswerItem[] answerItems =   questions.getSimpleQuestion().getAnswers().getAnswerItem();
         *//*

        for( AnswerItem  a : answerItems)
        {
            s += a.toString()+"\n";
            Type mediITemObjectType  = new TypeToken<MediaItem>(){}.getType();

            a.getExplain().getMedia().setMediaItemObject(gson.fromJson(a.get, mediITemObjectType));
        }*//*

        tv_json.setText(s);
        try{

        }

        catch (Exception e )
        {
          *//*  Log.d("JSON Exception", e.getMessage());
            tv_json.setText( e.getMessage());*//*
        }*/

    }
}

File: app/src/main/java/com/goldmines/xmlparsing/MainActivity.kt
package com.goldmines.xmlparsing

import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import android.support.v7.app.AppCompatActivity
import android.text.Html
import android.util.Log
import android.view.View
import android.widget.*
import com.goldmines.jsonkotlinpojos.AnswerItem
import com.goldmines.jsonkotlinpojos.MediaItem
import com.goldmines.jsonkotlinpojos.Question
import com.goldmines.xmlparsing.MediaX.desirialiser
import com.google.gson.Gson
import com.google.gson.GsonBuilder
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.activity_main2.*
import org.json.JSONObject
import org.json.XML


class MainActivity : AppCompatActivity() {
    var jsonObj: JSONObject? = null
    var questions :Question?=null
    var TAG:String?=null
    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        TAG = this.localClassName
        setSupportActionBar(toolbar)

        jsonObj = XML.toJSONObject(XMLFile.sampleXml)


       /* questions = Gson().fromJson(jsonObj.toString(), Question::class.java)*/

        questions=    GsonBuilder().registerTypeAdapter(MediaX::class.java, desirialiser).create().fromJson(jsonObj.toString(), Question::class.java)

/*        tv_json.text = questions.simpleQuestion.answers.answerItem[0].explain.media.mediaItem.location*/



        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
            genViews()
        }


    }

    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
    private fun genViews() {
        genQues()
        genAnswers()
    }

    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
    private fun genAnswers() {
        val rg = RadioGroup(this)

        var ansItemList = questions?.simpleQuestion?.answers?.answerItem!!
        var arrayString  = ""



        for(aI : AnswerItem in ansItemList)
        {

            if(aI.explain.media.mediaItemArray!=null)
            {

                for (  mI : MediaItem in aI.explain.media.mediaItemArray)
                {
                    arrayString += "Array" + mI.content+" \n"
                }

            }
            else if(aI.explain.media.mediaItemObject!=null)
            {
                arrayString +="Object" + aI.explain.media.mediaItemObject.content+" \n"
            }
            else {
                arrayString = "is null"
            }

            Log.d(TAG, "arrayStrin :"+arrayString)
            addRadioButton(aI, rg)
        }

        rl_option_container.addView(rg)

    }

    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
    private fun addRadioButton(aI : AnswerItem, rg: RadioGroup) {

        val stringArray = aI.location.split(",")
        var lP =stringArray[0].toInt()

        var  tP =stringArray[1].toInt()
        var w =stringArray[2].toInt()
        var  h =stringArray[3].toInt()



        var params =  LinearLayout.LayoutParams(w,h)
        params.leftMargin=lP;
        params.topMargin=tP




        val rdbtn = RadioButton(this)
        rdbtn.id = View.generateViewId()
        rdbtn.text =  Html.fromHtml(aI.content)
    rg.addView(rdbtn,params)
 /*      rg.addView(rdbtn)*/
     //   rg.addView(rdbtn)
    }







    private fun genQues() {
        var mediItemList = questions?.simpleQuestion?.media?.mediaItem

        if(mediItemList!=null)

        {
            for ( mI : MediaItem in mediItemList)
            {
                selectQuesType( mI)
            }
        }
        else{
            Log.d(TAG,"mediaItemList is not array")
        }

    }

    private fun selectQuesType(mI : MediaItem) {

        when(mI.signature){
            "Picture" -> createQuesIV(mI)
            "Rtf" -> createQuesTV( mI)

            else ->{

                Log.d(TAG, "signature does not match.")
            }
        }
    }

    private fun createQuesIV(mI: MediaItem) {

        var iv = ImageView(this)
        iv.setImageResource(android.R.mipmap.sym_def_app_icon)
        setParams(mI, iv)
    }

    private fun createQuesTV(mI: MediaItem) {
        var tvQues_1 = TextView(this)
        tvQues_1.text = Html.fromHtml(mI.content)
        tvQues_1.textSize = mI.size.toFloat();
        setParams(mI, tvQues_1)

    }

    private fun setParams(mI: MediaItem, view: View) {
        val stringArray = mI.location.split(",")
        var lP =stringArray[0].toInt()

        var  tP =stringArray[1].toInt()
        var w =stringArray[2].toInt()
        var  h =stringArray[3].toInt()



        var params =  LinearLayout.LayoutParams(w,h)
        params.leftMargin=lP;
        params.topMargin=tP

        ll_ques_container.addView(view,params)
    }


}



